prometheus/common/alert-rules.d/crunchy-alert-rules-etcd.yml.example

```diff
@@ -10,11 +10,58 @@ groups:
 
 ########## ETCD EXPORTER RULES ##########
 
+#### IMPORTANT NOTE: These alerts only support monitoring a single etcd cluster per prometheus instance ####
+
+# Ensure there is at least one etcd node reporting metrics
+  - alert: ETCDAbsent
+    expr: absent(etcd_server_has_leader)
+    for: 10s
+    labels:
+      service: etcd
+      severity: critical
+      severity_num: 300
+    annotations:
+      description: 'No etcd metric endpoints found'
+
+# Ensure there are an odd number of etcd nodes providing metrics
+  - alert: ETCDOddCount
+    expr: (count(etcd_server_has_leader) % 2) != 1
```

Comment:
So just for my understanding, the `etcd_server_has_leader` is a metric that is returned from each etcd endpoint that is being monitored which should be true in a healthy cluster for each node—hence this equates to the number of nodes, roughly speaking?

Reply:
Yes, the metric is always returned, even if there isn't a leader. The value just changes then